src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -97,17 +112,48 @@ private static NETCore.Client.DumpType MapDumpType(DumpType dumpType)
             {
                 case DumpType.Full:
                     return NETCore.Client.DumpType.Full;
-                case DumpType.MiniWithHeap:
+                case DumpType.WithHeap:
                     return NETCore.Client.DumpType.WithHeap;
                 case DumpType.Triage:
                     return NETCore.Client.DumpType.Triage;
-                case DumpType.Normal:
+                case DumpType.Mini:
                     return NETCore.Client.DumpType.Normal;
                 default:
                     throw new ArgumentException("Unexpected dumpType", nameof(dumpType));
             }
         }
 
+        private int GetSingleProcessId()
+        {
+            // Short-circuit for when running in a container on Kubernetes where the entrypoint
+            // of the container is a dotnet application.
+            if (RuntimeInfo.IsKubernetes && null != Process.GetProcessById(1))
```

Comment:
>IsKubernetes [](start = 28, length = 12)

This shouldn't be IsKubernetes, but rather IsDocker.

Reply:
I'll have to see if we can detect that it's running in Docker.